File: scripts/02_process_data.R
# 02_process_data.R --------------------------------------------------------- #
# Author: Louis Trocellier
# Description: Script for step 02 - Processing data.

# Execution ----------------------------------------------------------------- #
log_info("Starting step 02: Process data")

# 1. Load parameters/settings
settings <- process_data_settings()

# 2. Process data
processed_data <- process_data_execution(settings)

# 3. Export processed data to intermediate storage
process_data_exports(processed_data)

log_info("Step 02 completed successfully")

File: scripts/02_process_data.py
# scripts/02_process_data.py ---------------------------------------------- #
# Author: Louis Trocellier
# Description: Step 02 - Processing data.

from src.functions.f02_process_data import (
    process_load_raw_data,
    process_data,
    process_export_data
)

if __name__ == "__main__":
    # 1. Load raw data
    raw_data = process_load_raw_data()

    # 2. Process data
    processed_data = process_data(raw_data)

    # 3. Export processed data
    process_export_data(processed_data)